docs/tutorials/quantum_reinforcement_learning.ipynb

```diff
@@ -0,0 +1,4085 @@
+      "Episode 433/500, rewards 500.0\n",
+      "Episode 434/500, rewards 500.0\n",
+      "Episode 435/500, rewards 500.0\n",
+      "Episode 436/500, rewards 500.0\n",
+      "Episode 437/500, rewards 192.0\n",
+      "Episode 438/500, rewards 231.0\n",
+      "Episode 439/500, rewards 204.0\n",
+      "Episode 440/500, rewards 500.0\n",
+      "Episode 441/500, rewards 219.0\n",
+      "Episode 442/500, rewards 200.0\n",
+      "Episode 443/500, rewards 500.0\n",
+      "Episode 444/500, rewards 140.0\n",
+      "Episode 445/500, rewards 500.0\n",
+      "Episode 446/500, rewards 210.0\n",
+      "Episode 447/500, rewards 165.0\n",
+      "Episode 448/500, rewards 208.0\n",
+      "Episode 449/500, rewards 500.0\n",
+      "Episode 450/500, rewards 220.0\n",
+      "Episode 451/500, rewards 500.0\n",
+      "Episode 452/500, rewards 155.0\n",
+      "Episode 453/500, rewards 225.0\n",
+      "Episode 454/500, rewards 215.0\n",
+      "Episode 455/500, rewards 178.0\n",
+      "Episode 456/500, rewards 371.0\n",
+      "Episode 457/500, rewards 195.0\n",
+      "Episode 458/500, rewards 500.0\n",
+      "Episode 459/500, rewards 165.0\n",
+      "Episode 460/500, rewards 500.0\n",
+      "Episode 461/500, rewards 500.0\n",
+      "Episode 462/500, rewards 142.0\n",
+      "Episode 463/500, rewards 210.0\n",
+      "Episode 464/500, rewards 189.0\n",
+      "Episode 465/500, rewards 178.0\n",
+      "Episode 466/500, rewards 218.0\n",
+      "Episode 467/500, rewards 146.0\n",
+      "Episode 468/500, rewards 207.0\n",
+      "Episode 469/500, rewards 147.0\n",
+      "Episode 470/500, rewards 500.0\n",
+      "Episode 471/500, rewards 162.0\n",
+      "Episode 472/500, rewards 500.0\n",
+      "Episode 473/500, rewards 138.0\n",
+      "Episode 474/500, rewards 147.0\n",
+      "Episode 475/500, rewards 225.0\n",
+      "Episode 476/500, rewards 164.0\n",
+      "Episode 477/500, rewards 186.0\n",
+      "Episode 478/500, rewards 225.0\n",
+      "Episode 479/500, rewards 186.0\n",
+      "Episode 480/500, rewards 189.0\n",
+      "Episode 481/500, rewards 500.0\n",
+      "Episode 482/500, rewards 252.0\n",
+      "Episode 483/500, rewards 500.0\n",
+      "Episode 484/500, rewards 500.0\n",
+      "Episode 485/500, rewards 500.0\n",
+      "Episode 486/500, rewards 258.0\n",
+      "Episode 487/500, rewards 500.0\n",
+      "Episode 488/500, rewards 500.0\n",
+      "Episode 489/500, rewards 175.0\n",
+      "Episode 490/500, rewards 378.0\n",
+      "Episode 491/500, rewards 500.0\n",
+      "Episode 492/500, rewards 203.0\n",
+      "Episode 493/500, rewards 500.0\n",
+      "Episode 494/500, rewards 500.0\n",
+      "Episode 495/500, rewards 500.0\n",
+      "Episode 496/500, rewards 211.0\n",
+      "Episode 497/500, rewards 500.0\n",
+      "Episode 498/500, rewards 500.0\n",
+      "Episode 499/500, rewards 500.0\n",
+      "Episode 500/500, rewards 233.0\n"
+     ]
+    }
+   ],
+   "source": [
```

Comment:
I broke down both training cells into helper functions that should make them less heavy to process / easier to read. I don't see the point in breaking down the training loop even more. The REINFORCE implementation of TF-Agents made sense because they implemented things in a way that was compatible with DDPG, TD3, PPO, SAC (where data collection is naturally separated from training) but in our case there's really no point, things are simple enough to leave them as they are.